File: app/src/main/java/com/example/ase_project_nutrismart/Model/SelectedGroceryModel.java
package com.example.ase_project_nutrismart.Model;

import androidx.room.ColumnInfo;
import androidx.room.PrimaryKey;

import com.example.ase_project_nutrismart.Response.Grocery;

import java.util.ArrayList;

public class SelectedGroceryModel {
    @PrimaryKey(autoGenerate = true)
    private int id;

    public ArrayList<Grocery> getData() {
        return data;
    }

    public void setData(ArrayList<Grocery> data) {
        this.data = data;
    }

    @ColumnInfo(name="data")
    public ArrayList<Grocery> data;

    public String getPurchasedDate() {
        return purchasedDate;
    }

    public void setPurchasedDate(String purchasedDate) {
        this.purchasedDate = purchasedDate;
    }

    public String getUserEmail() {
        return userEmail;
    }

    public void setUserEmail(String userEmail) {
        this.userEmail = userEmail;
    }

    @ColumnInfo(name="purchaseDate")
    public String purchasedDate;

    @ColumnInfo(name="userEmail")
    public String userEmail;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: app/src/main/java/com/example/ase_project_nutrismart/Model/SelectedGroceryModel.kt
package com.example.ase_project_nutrismart.Model

import androidx.room.PrimaryKey
import com.example.ase_project_nutrismart.Response.Grocery
import androidx.room.ColumnInfo
import java.util.ArrayList

class SelectedGroceryModel {
    @PrimaryKey(autoGenerate = true)
    var id = 0

    @ColumnInfo(name = "data")
    var data: ArrayList<Grocery>? = null

    @JvmField
    @ColumnInfo(name = "purchaseDate")
    var purchasedDate: String? = null

    @JvmField
    @ColumnInfo(name = "userEmail")
    var userEmail: String? = null
}